src/libraries/System.Net.Http/tests/UnitTests/Headers/HttpRequestHeadersTest.cs

```diff
@@ -1553,6 +1553,24 @@ public void Pragma_UseAddMethodWithInvalidValue_InvalidValueRecognized()
 
         #endregion
 
+        [Fact]
+        public void NonValidatedHeader_Orders()
```

Comment:
```suggestion
        public void TryAddWithoutValidation_ValidAndInvalidValues_InsertionOrderIsPreserved()
```

Please move this test to the more general `HttpHeadersTest` file. We will also want to add more tests for both known and custom headers.